custom_components/ai_thermostat/climate.py

```diff
@@ -70,823 +66,778 @@
+		)
+		async_track_state_change_event(
+				self.hass, [self.heater_entity_id], self._async_trv_changed
+		)
+		if self.window_sensors_entity_ids:
+			async_track_state_change_event(
+					self.hass, [self.window_sensors_entity_ids], self._async_window_changed
+			)
+		
+		if self.night_temp != -1:
+			async_track_time_change(
+					self.hass,
+					self._async_timer_trigger,
+					self.night_start.hour,
+					self.night_start.minute,
+					self.night_start.second,
+			)
+			async_track_time_change(
+					self.hass,
+					self._async_timer_trigger,
+					self.night_end.hour,
+					self.night_end.minute,
+					self.night_end.second,
+			)
+		
+		@callback
+		def _async_startup(*_):
+			"""Init on startup."""
+			
+			_LOGGER.info("Starting ai_thermostat for %s with version: 0.8.6 waiting for entity to be ready...", self.name)
+			
+			loop = asyncio.get_event_loop()
+			loop.create_task(self.startup())
+		
+		if self.hass.state == CoreState.running:
+			_async_startup()
+		else:
+			self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
+		
+		# Check If we have an old state
+		old_state = await self.async_get_last_state()
+		if old_state is not None:
+			# If we have no initial temperature, restore
+			if self._target_temp is None:
+				# If we have a previously saved temperature
+				if old_state.attributes.get(ATTR_TEMPERATURE) is None:
+					self._target_temp = self.min_temp
+					_LOGGER.debug(
+							"Undefined target temperature, falling back to %s",
+							self._target_temp,
+					)
+				else:
+					self._target_temp = float(old_state.attributes[ATTR_TEMPERATURE])
+			if not self._hvac_mode and old_state.state:
+				self._hvac_mode = old_state.state
+		
+		else:
+			# No previous state, try and restore defaults
+			if self._target_temp is None:
+				self._target_temp = self.min_temp
+			_LOGGER.debug(
+					"No previously saved temperature, setting to %s", self._target_temp
+			)
+		
+		# Set default state to off
+		if not self._hvac_mode:
+			self._hvac_mode = HVAC_MODE_OFF
+	
+	async def startup(self):
+		await asyncio.sleep(5)
+		
+		while self.startup_running:
+			sensor_state = self.hass.states.get(self.sensor_entity_id)
+			trv_state = self.hass.states.get(self.heater_entity_id)
+			
+			if sensor_state is None:
+				_LOGGER.error("ai_thermostat %s temperature sensor: %s is not in HA or wrong spelled", self.name, self.sensor_entity_id)
+				return False
+			if trv_state is None:
+				_LOGGER.error("ai_thermostat %s TRV: %s is not in HA or wrong spelled", self.name, self.heater_entity_id)
+				return False
+			if self.window_sensors_entity_ids:
+				window = self.hass.states.get(self.window_sensors_entity_ids)
+				
+				if window is None:
+					_LOGGER.error("ai_thermostat %s window sensor: %s is not in HA or wrong spelled", self.name, self.window_sensors_entity_ids)
+					return False
+			
+			_ready = True
+			
+			if sensor_state.state in (STATE_UNAVAILABLE, STATE_UNKNOWN):
```

Comment:
Should contain also None, because some devices will return none in state.

Reply:
So the devices might return either `sensor_state` as None or `sensor_state.state` as None?

Reply:
https://github.com/KartoffelToby/ai_thermostat/pull/115/commits/9b2893aa077f195088f37fe0949daf7224a4431a schould fix that.

Reply:
Yes, sensor_state is the wrong name for this var, if sensor_state is None, its wrong spelled or not in HA, if sensor_state.state is None the sensor is maybe not ready yet